src/Sprite/OP2BmpLoader.cpp

```diff
@@ -35,19 +36,17 @@ void OP2BmpLoader::ExtractImage(std::size_t index, const std::string& filenameOu
 
 std::vector<Color> OP2BmpLoader::GetPalette(const ImageMeta& imageMeta)
 {
-	std::vector<Color> palette;
+	uint16_t bitCount;
 
-	if (imageMeta.type.isShadow)
-	{
-		// Shadow sprites use a 2 color (1 bit) palette
-		palette.resize(sizeof(Palette1Bit));
+	if (imageMeta.type.isShadow) {
+		bitCount = 1;
 	}
-	else
-	{
-		// All other sprites use a 256 color (8 bit) palette
-		palette.resize(sizeof(Palette8Bit));
+	else {
+		bitCount = 8;
 	}
```

Comment:
You can potentially collapse this down to a single line by using the ternary operator. That would also allow the variable to be marked with `const`:
```cpp
const uint16_t bitCount = imageMeta.type.isShadow ? 1 : 8;
```